File: process/OFM/tidy_ofm_apr_inter.py
# This script will tidy and compile downloaded intercensal OFM April 1 data
import os
from functools import partial, reduce

exec(open('functions_tidy.py').read())

data_dir = '../../data'
filename = 'ofm_april1_population_intercensal.xlsx'

# Read in entire workbook as dictionary
all = pd.read_excel(os.path.join(data_dir, filename), sheet_name=None)
del all['Readme']

# tidy all dataframes in the workbook
df_all = list()
for tabname in list(all.keys()):
    temp_df = tidy_ofm_apr(all[tabname], tabname)
    df_all.append(temp_df)

# join all dataframes into one
join_dfs = partial(pd.merge, on = ['county', 'jurisdiction', 'estimate_year'])
df = reduce(join_dfs, df_all)

File: archive/process/OFM/tidy_ofm_apr_inter.R
# This script will tidy and compile downloaded intercensal OFM April 1 data

source(here::here('process', 'OFM', 'config.R'))
source(here::here('process', 'OFM', 'functions_tidy.R'))

data_dir <- 'data'
filename <- filename_inter

tabs <- getSheetNames(here(data_dir, filename))
tabs <- tabs[which(tabs != 'Readme')]
all <- map(tabs, ~read.xlsx(xlsxFile = here(data_dir, filename), sheet = .x))

names(all) <- tabs

# tidy all dataframes in the workbook
# join all dataframes into one
join_dfs <- partial(merge, on = c('county', 'jurisdiction', 'estimate_year'))
df_inter <- map2(all, tabs, ~tidy_ofm_apr(.x, .y)) %>% 
  reduce(join_dfs)

rm(all)

